### ISSUE TITLE
On tor browser does not seem to work 

### ISSUE THREAD

**Issue by xzj8b3** [keepassxreboot/keepassxc-browser#687]
On Tor, the browser does not seem to work and does not connect to the database.

**Comment by xzj8b3** [comment]
KeePassXC-Browser versione: 1.5.3

**Comment by xzj8b3** [comment]
The key exchange failed.

**Comment by xzj8b3** [comment]
Cannot connect to KeePassXC. Check that browser integration is enabled in KeePassXC settings.

**Comment by droidmonkey** [comment]
Need way more info. Don't delete the issue template.

**Comment by xzj8b3** [comment]
> Need way more info. Don't delete the issue template.

translators unfortunately are not the top, I have not understood what you mean....

**Comment by varjolintu** [comment]
@xzj8b3 When you are making a new issue, there's some text ready for you to fill. Now you just deleted all the extra info that could have been helpful, for example your operating system and KeePassXC version.

**Comment by xzj8b3** [comment]
however with tor browser does not go

**Comment by xzj8b3** [comment]
> > Need way more info. Don't delete the issue template.
> 
> translators unfortunately are not the top, I have not understood what you mean....

however with tor browser does not go

**Comment by xzj8b3** [comment]
> You can check for the needed. json file from these paths:
> Windows: `HKEY_CURRENT_USER\\Software\\Mozilla\\NativeMessagingHosts\\org.keepassxc.keepassxc_browser`
> macOS: `/Library/Application Support/TorBrowser-Data/Browser/Mozilla/NativeMessagingHosts"`
> Linux: `/.tor-browser/app/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts`

"allowed_extensions": [
        "keepassxc-browser@keepassxc.org"
    ],
    "description": "KeePassXC integration with native messaging support",
    "name": "org.keepassxc.keepassxc_browser",
    "path": "/usr/bin/keepassxc-proxy",
    "type": "stdio"
}

Torbrowser-launcher is used on Fedora, but I don't think it changes anything.

**Comment by varjolintu** [comment]
Is there `/usr/bin/keepassxc-proxy` file? I'm not sure how the Tor Browser is bundled with Fedora. is it

**Comment by xzj8b3** [comment]
> > You can check for the needed. json file from these paths:
> > Windows: `HKEY_CURRENT_USER\\Software\\Mozilla\\NativeMessagingHosts\\org.keepassxc.keepassxc_browser`
> > macOS: `/Library/Application Support/TorBrowser-Data/Browser/Mozilla/NativeMessagingHosts"`
> > Linux: `/.tor-browser/app/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts`
> 
> "allowed_extensions": [
> "[keepassxc-browser@keepassxc.org](mailto:keepassxc-browser@keepassxc.org)"
> ],
> "description": "KeePassXC integration with native messaging support",
> "name": "org.keepassxc.keepassxc_browser",
> "path": "/usr/bin/keepassxc-proxy",
> "type": "stdio"
> }
> 
> Torbrowser-launcher is used on Fedora, but I don't think it changes anything.

in casa no

**Comment by xzj8b3** [comment]
> > > You can check for the needed. json file from these paths:
> > > Windows: `HKEY_CURRENT_USER\\Software\\Mozilla\\NativeMessagingHosts\\org.keepassxc.keepassxc_browser`
> > > macOS: `/Library/Application Support/TorBrowser-Data/Browser/Mozilla/NativeMessagingHosts"`
> > > Linux: `/.tor-browser/app/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts`
> > 
> > 
> > "allowed_extensions": [
> > "[keepassxc-browser@keepassxc.org](mailto:keepassxc-browser@keepassxc.org)"
> > ],
> > "description": "KeePassXC integration with native messaging support",
> > "name": "org.keepassxc.keepassxc_browser",
> > "path": "/usr/bin/keepassxc-proxy",
> > "type": "stdio"
> > }
> > Torbrowser-launcher is used on Fedora, but I don't think it changes anything.
> 
> in casa no

Yes, there is.

**Comment by lotth** [comment]
Hi,
I can't make tor browser connect to KeepassXC either, on Fedora as well (Fedora 30). Connexion works on Firefox with the same native-messaging-hosts in ~/.mozilla instead of ~/.tor-browser/app/Browser/TorBrowser/Data/Browser

My KeepassXC version is 2.5.1.

My org.keepassxc.keepassxc_browser.json:
    "allowed_extensions": [
        "keepassxc-browser@keepassxc.org"
    ],
    "description": "KeePassXC integration with native messaging support",
    "name": "org.keepassxc.keepassxc_browser",
    "path": "/usr/bin/keepassxc-proxy",
    "type": "stdio"
}

about:debuggin log:
Error 5: Cannot connect to KeePassXC. Check that browser integration is enabled in KeePassXC settings.

Tor browser integration is selected in KeePassXC preferences

**Comment by xzj8b3** [comment]
> @xzj8b3 When you are making a new issue, there's some text ready for you to fill. Now you just deleted all the extra info that could have been helpful, for example your operating system and KeePassXC version.
> 
> We **need** that info.

keepassxc-2.5.1-1.fc31.src.rpm
Fedora 31 x86-64

**Comment by xzj8b3** [comment]
> Is there `/usr/bin/keepassxc-proxy` file? I'm not sure how the Tor Browser is bundled with Fedora.

![Impostazioni - KeePassXC_001](https://user-images.githubusercontent.com/26720956/69496831-47ddee00-0ece-11ea-8212-417c07634084.png)

does not work

**Comment by varjolintu** [comment]
Used some `strace`  magic to see where the issue is. For this, I used the official Tor Browser package https://www.torproject.org/dist/torbrowser/9.0.1/tor-browser-linux64-9.0.1_en-US.tar.xz

Starting that seems to look at only these two locations:
- `/home/varjolintu/tor-browser_en-US/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts/org.keepassxc.keepassxc_browser.json`
- `/usr/lib/mozilla/native-messaging-hosts/org.keepassxc.keepassxc_browser.json`

For a workaround, copy the script file under `.mozilla` to one of those directories. Seems like at some point Tor Browser has stopped to look under the user's configuration paths, and uses its own instead, like a portable application.

**Comment by xzj8b3** [comment]
> For a workaround, copy the script file under `.mozilla/native-messaging-hosts/` to one of those directories. Seems like at some point Tor Browser has stopped to look under the user's configuration paths, and uses its own instead, like a portable application.

Yes but correct it, I do not care Tor Browser that I do not have between repositories .... Torbrowser-launcer on fedora and other distributions is updated regularly and those who have installed it already has a configuration on that browser!!!!!!

If every project is so poor I'll go back to lastpass or others!!!!

**Comment by varjolintu** [comment]
Well, at this point it's hard to correct because if everyone has a different location for the configuration files, we must first implement a way to use a Custom Browser in the Browser Integration settings.

I'm not sure what you meant, but I guess Torbrowser-launcher uses the user's configuration paths normally.

**Comment by varjolintu** [comment]
I'll install Fedora 31 and try to find out where the problem is.

**Comment by xzj8b3** [comment]
> I'll install Fedora 31 and try to find out where the problem is.

Thank you better

**Comment by varjolintu** [comment]
Installed a fresh Fedora 31 with Torbrowser-launcher, and it seems it's trying to load the needed. json file from the following path: `/home/varjolintu/.local/share/torbrowser/tbb/x86_64/tor-browser_en-US/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts/org.keepassxc.keepassxc_browser.json`. So if you manually copy the script to that location, I suppose everything should work.

Of course this makes fixing this much harder, because not everyone uses Torbrowser-launcher, plus there's a localization string in the path.

**Comment by lotth** [comment]
> Used some `strace` magic to see where the issue is. For this, I used the official Tor Browser package https://www.torproject.org/dist/torbrowser/9.0.1/tor-browser-linux64-9.0.1_en-US.tar.xz
> 
> Starting that seems to look at only these two locations:
> 
>     * `/home/varjolintu/tor-browser_en-US/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts/org.keepassxc.keepassxc_browser.json`
> 
>     * `/usr/lib/mozilla/native-messaging-hosts/org.keepassxc.keepassxc_browser.json`
> 
> 
> For a workaround, copy the script file under `.mozilla/native-messaging-hosts/` to one of those directories. Seems like at some point Tor Browser has stopped to look under the user's configuration paths, and uses its own instead, like a portable application.

Hurray! Thanks varjolintu. I copied native-messaging-hosts in <tor browser install path>/Browser/TorBrowser/Data/Browser/.mozilla and I get a successful connection.
May I suggest that you add an advice about that in KeepassXC documentation? Even if tor browser install path may differ between different distribs, it is easy to find the .mozilla folder.

**Comment by varjolintu** [comment]
@lotth I can add that to the troubleshootin wiki for now.
